gdx/src/com/badlogic/gdx/files/FileHandle.java

```diff
@@ -416,25 +401,48 @@ public void writeBytes (byte[] bytes, int offset, int length, boolean append) {
 		}
 	}
 
-	/** Returns the paths to the children of this directory. Returns an empty list if this file handle represents a file and not a
-	 * directory. On the desktop, an {@link FileType#Internal} handle to a directory on the classpath will return a zero length
-	 * array.
-	 * @throws GdxRuntimeException if this file is an {@link FileType#Classpath} file. */
-	public FileHandle[] list () {
-		if (type == FileType.Classpath) throw new GdxRuntimeException("Cannot list a classpath directory: " + file);
-		String[] relativePaths = file().list();
-		if (relativePaths == null) return new FileHandle[0];
-		FileHandle[] handles = new FileHandle[relativePaths.length];
-		for (int i = 0, n = relativePaths.length; i < n; i++)
-			handles[i] = child(relativePaths[i]);
-		return handles;
-	}
-
-	/** Returns the paths to the children of this directory that satisfy the specified filter. Returns an empty list if this file
-	 * handle represents a file and not a directory. On the desktop, an {@link FileType#Internal} handle to a directory on the
-	 * classpath will return a zero length array.
-	 * @param filter the {@link FileFilter} to filter files
-	 * @throws GdxRuntimeException if this file is an {@link FileType#Classpath} file. */
+	 /**
+	  * Returns the paths to the children of this directory. Returns an empty list if this file handle represents a file and not a
+	  * directory. An {@link FileType#Internal} handle to a directory on the classpath will search for
+	  * an asset.index file in the * root of the asset path. This index should be generated at compile time by a gradle task.
+	  * See for reference: * https://gist.github.com/abueide/e86ea0111025233fb6493dfbcc46501e

```

Comment:
Referencing a gist seems messy. It would be much better to document the feature fully here, including the full description of the `asset.index` file format (including where the file should be, what directory separators it should contain, if it should contain leading `/`, ...), so that somebody who would want to generate it in a different build system or through a shell script would not have to reverse engineer the source code.

Reply:
Yeah the gist was just a stop-gap until I knew if this would be an accepted change or not. I was planning on editing the section on the libgdx wiki and then linking to that. Writing a spec in the doc instead of just a gradle task sounds like a good idea though.